client/widgets/MiscWidgets.cpp

```diff
@@ -357,6 +382,55 @@ CTownTooltip::CTownTooltip(Point pos, const CGTownInstance * town)
 	init(InfoAboutTown(town, true));
 }
 
+CInteractableTownTooltip::CInteractableTownTooltip(Point pos, const CGTownInstance * town)
+		: CGarrisonInt(pos.x, pos.y+73, 4, Point(0, 0), town->getUpperArmy(), nullptr, true, true, CGarrisonInt::EGarrisonIntSlotsLayout::REVERSED_TWO_ROWS)
+{
+	init(InfoAboutTown(town, true));
+}
+
+void CInteractableTownTooltip::init(const InfoAboutTown & town)
```

Comment:
Perhaps try to reuse common code from non-interactive tooltip to avoid yet another copy-paste of town icon / fort / hall / income? Same question about hero tooltip

Not sure if this is doable, so feel free to ignore

Reply:
Should be doable by multiple inheritance, but I would rather focus on looking into other features currently